controller/AdminAccessController.php

```diff
@@ -163,13 +210,53 @@ public function savePermissions()
      */
     protected function validatePermissions($usersPrivileges)
     {
-        $pp = new PermissionProvider();
+        $pp = $this->getPermissionProvider();

```

Comment:
$pp should be $permissionProvider. Don't use these short variable names please! ( $i is okay in for loop :) )